app/services/container_dashboard_service.rb

```diff
@@ -63,27 +63,33 @@ def status
   end
 
   def providers
-    [
-      {
-        :iconClass    => "pficon pficon-openshift",
-        :count        => ManageIQ::Providers::Openshift::ContainerManager.count +
-                         ManageIQ::Providers::OpenshiftEnterprise::ContainerManager.count,
-        :id           => "openshift",
-        :providerType => "OpenShift"
-      },
-      {
-        :iconClass    => "pficon pficon-kubernetes",
-        :count        => ManageIQ::Providers::Kubernetes::ContainerManager.count,
-        :id           => "kubernetes",
-        :providerType => "Kubernetes"
-      },
+    provider_types = {
+      :openshift  => [
+        ManageIQ::Providers::Openshift::ContainerManager,
+        ManageIQ::Providers::OpenshiftEnterprise::ContainerManager
+      ],
+      :kubernetes => [
+        ManageIQ::Providers::Kubernetes::ContainerManager
+      ],
+      :atomic     => [
+        ManageIQ::Providers::Atomic::ContainerManager,
+        ManageIQ::Providers::AtomicEnterprise::ContainerManager
+      ]
+    }
+
+    provider_types.collect { |icon_type, types|
```

Comment:
It seems to me that instead of doing 5 queries (`count`), you could do:

    ems = ManageIQ::Providers::ContainerManager.group(:type).count.select do |k|
      Vmdb::PermissionStores.instance.supported_ems_type?(k.constantize.ems_type)
    end

and then you'll use `ems` to build the hash list you need.